test-suite/test-squid-conf.sh

```diff
@@ -68,8 +90,34 @@ then
             exit 1;
         fi
     done < $instructionsFile
+fi
+
+errorLog="squid-stderr.log"
 
-    # TODO: Add support for the "require-failure" instruction.
+$sbindir/squid -k parse -f $configFile 2> $errorLog
+result=$?
+
+if test -z "$failureRegex"
+then
+    # a positive test does not require special $result interpretation
+    exit $?
 fi
 
-exec $sbindir/squid -k parse -f $configFile
+if test "$result" -eq 0
+then
+    echo "ERROR: Squid successfully parsed malformed $configFile instead of rejecting it"
+    exit 1;
+fi
+
+if ! grep -q -E "$failureRegex" $errorLog
```

Comment:
awk would be a better tool for matching here as it can be given rules for multi-line matches.

Reply:
Please feel free to switch to awk, here or in a follow up PR (or when it actually becomes needed). I do not know enough about awk, and do not see the need to switch to awk now to implement that switch myself.